### PR TITLE
Consolidate all metadata Get's and Set's to central functions.

### PR DESCRIPTION
**Pull request by dragotin**
In the decomposed FS, access to xattr was spread all over. This
patch consolidates that to use either the Node.SetMetadata() or
xattrs.Set(). This allows us to hook in indexing for example.

This does not yet handle Remove, List and others. Subject of another patch.

### CODE REVIEW

File: pkg/storage/utils/decomposedfs/xattrs/xattrs.go [hunk 1]
```diff
@@ -127,3 +128,55 @@ func CopyMetadata(s, t string, filter func(attributeName string) bool) error {
 	}
 	return nil
 }
+
+// The primitive function to set the extended attribute
```
[the remark this answers is not in the document]
Comment (dragotin) [reply pkg/storage/utils/decomposedfs/xattrs/xattrs.go:132]: @aduffeck FYI

**Comment by dragotin** [comment]
Get, Set and All also give us a central point to deal with locking, indexing, plausability checks and logging.

File: pkg/storage/utils/decomposedfs/node/node.go [hunk 2]
```diff
@@ -127,47 +126,42 @@ func (n *Node) ChangeOwner(new *userpb.UserId) (err error) {
...
+}
+
+// WriteAllNodeMetadata writes the Node metadata to disk
+func (n *Node) WriteAllNodeMetadata(owner *userpb.UserId) (err error) {
+	attribs := make(map[string]string)
+
+	attribs[xattrs.ParentidAttr] = n.ParentID
+	attribs[xattrs.NameAttr] = n.Name
+	attribs[xattrs.BlobIDAttr] = n.BlobID
+	attribs[xattrs.BlobsizeAttr] = fmt.Sprintf("%d", n.Blobsize)
```
Comment (C0rby) [inline pkg/storage/utils/decomposedfs/node/node.go:151]:
```suggestion
	attribs[xattrs.BlobsizeAttr] = strconv.FormatInt(n.Blobsize, 10)
```

File: pkg/storage/utils/decomposedfs/xattrs/xattrs.go [hunk 3]
```diff
@@ -127,3 +128,57 @@ func CopyMetadata(s, t string, filter func(attributeName string) bool) error {
...
+		name := attrNames[i]
+		val, err := xattr.Get(filePath, name)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to read extended attrib")
+		}
+		attribs[name] = string(val)
+	}
+
+	return attribs, nil
+}
```
Comment (C0rby) [inline pkg/storage/utils/decomposedfs/xattrs/xattrs.go:184]:
```suggestion
func All(filePath string) (map[string]string, error) {
	attrNames, err := xattr.List(filePath)
	if err != nil {
		return nil, errors.Wrap(err, "xattrs: Can not list extended attributes")
	}

       attribs := make(map[string]string, len(attrNames))
	for _, name := range attrNames {
		val, err := xattr.Get(filePath, name)
		if err != nil {
			return nil, errors.Wrap(err, "Failed to read extended attrib")
		}
		attribs[name] = string(val)
	}

	return attribs, nil
}
```

**Review by C0rby — CHANGES_REQUESTED** [review]

**Review by C0rby — APPROVED** [review]

**Pull request merged**